File: go-advanced/pkg/utils/validator.go
package utils

import "regexp"

type Validator struct{}

func (v *Validator) IsValidEmail(email string) bool {
	emailRegex := regexp.MustCompile(`^[^\s@]+@[^\s@]+\.[^\s@]+$`)
	return emailRegex.MatchString(email)
}

func (v *Validator) IsValidName(name string) bool {
	return len(name) >= 2 && len(name) <= 100
}

var ValidatorInstance = &Validator{}

File: python-advanced/src/utils/validator.py
import re

class Validator:
    @staticmethod
    def is_valid_email(email: str) -> bool:
        email_regex = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
        return bool(re.match(email_regex, email))

    @staticmethod
    def is_valid_name(name: str) -> bool:
        return name and 2 <= len(name) <= 100
